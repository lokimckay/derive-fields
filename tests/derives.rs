mod common;

use common::*;
use derive_fields::{FieldKeys, Fields};

#[derive(Fields, FieldKeys)]
#[fields_derives(Debug)]
#[field_keys_derives(Debug)]
pub struct ExampleStruct {
    _name: String,
    _category: Category,
    _really_really_long_key: bool,
}

fn main() {
    // Confirm that key and field no longer implement clone
    let key = ExampleStructFieldKey::Name;
    let field = ExampleStructField::Name("Albert".to_string());
    let _ = key.clone();
    let _ = field.clone();
}
