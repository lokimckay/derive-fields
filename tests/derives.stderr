error[E0599]: no method named `clone` found for enum `ExampleStructFieldKey` in the current scope
  --> tests/derives.rs:19:17
   |
 6 | #[derive(Fields, FieldKeys)]
   |                  --------- method `clone` not found for this enum
...
19 |     let _ = key.clone();
   |                 ^^^^^ method not found in `ExampleStructFieldKey`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `clone`, perhaps you need to implement it:
           candidate #1: `Clone`

error[E0599]: no method named `clone` found for enum `ExampleStructField` in the current scope
  --> tests/derives.rs:20:19
   |
 6 | #[derive(Fields, FieldKeys)]
   |          ------ method `clone` not found for this enum
...
20 |     let _ = field.clone();
   |                   ^^^^^ method not found in `ExampleStructField`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `clone`, perhaps you need to implement it:
           candidate #1: `Clone`
